crawl-ref/source/spl-cast.cc

```diff
@@ -728,9 +728,18 @@ bool cast_a_spell(bool check_range, spell_type spell)
                 }
                 else
                 {
-                    mprf(MSGCH_PROMPT, "Casting: <w>%s</w>",
-                         spell_title(you.last_cast_spell));
-                    mprf(MSGCH_PROMPT, "Confirm with . or Enter, or press ? or * to list all spells.");
+                    ostringstream desc;
+                    const string failure = failure_rate_to_string(raw_spell_fail(you.last_cast_spell));
+                    int colour = failure_rate_colour(you.last_cast_spell);
+                    desc << "<" << colour_to_str(colour) << ">";
+                    desc << chop_string(failure, failure.length());
+                    desc << "</" << colour_to_str(colour) << ">";
+                    mprf(MSGCH_PROMPT, "Casting: <w>%s "
+                                       "(%s risk of failure)</w>",
+                                       spell_title(you.last_cast_spell),
+                                       desc.str().c_str());
+                    mprf(MSGCH_PROMPT, "Confirm with . or Enter, or press "
+                                       "? or * to list all spells.");
                 }
```

Comment:
If you wanted to clean this up more I'd probably build the colorization stuff into `failure_rate_to_string` with a boolean that enables or disables it (give it a default of false to have the same behavior as currently).

Reply:
I was looking into this. `failure_rate_to_string` takes an int, not a spell. So if I just add a boolean to enable/disable colourization, I will still need to pass the color (or pass a spell so I can call `failure_rate_color()`.

But if I'm just going to pass a spell or a colour, can't I just use that as the indicator that I want the string colourized? (i.e. if I update it to: `failure_rate_to_string(int failure, int colour)` and just set the default colour to COLOUR_UNDEF, so if `colour == COLOUR_UNDEF`, don't colourize.

alternatively, I could update the function to pass the spell and the enable/disable colourization.  Then I could call the appropriate functions inside that routine (`raw_spell_fail` and `failure_rate_to_colour`) directly.  This wouldn't match the way the other `failure_rate_to_foo()` routines are set up, so I'd have to change those too.  But they all, but one, pass `raw_spell_fail(spell)`, so seems like it would simplify things.  I'm assuming, though, that it's set up like it is now for a reason.

Oh and the one location that uses `failure_rate_to_string()` that isn't a spell is:
```
string describe_talent(const talent& tal)
{
    ASSERT(tal.which != ABIL_NON_ABILITY);

    const string failure = failure_rate_to_string(tal.fail)
        + (testbits(get_ability_def(tal.which).flags, abflag::hostile)
           ? " hostile" : "");
```